cpp/src/shim/cufile.cpp

```diff
@@ -37,27 +37,25 @@ cuFileAPI::cuFileAPI()
     CUfileError_t const error = GetVersion(&ver);
     if (error.err == CU_FILE_SUCCESS) { version = ver; }
   } catch (std::runtime_error const&) {
+    version = 1070;
   }
+#else
+  version = 1070;
 #endif
```

Comment:
To be consistent with previous behavior, we don't need to set the version number here in the else branch. It will take the default value of 0 (specified in `cuFileAPI`'s constructor). I think we could add back some of the comments related to the version checking API. So I'd suggest this change:
```suggestion
  // Note: `version` is 0 for cuFile versions prior to v1.8 because `cuFileGetVersion`
  // did not exist.
#ifdef KVIKIO_CUFILE_VERSION_API_FOUND
  try {
    get_symbol(GetVersion, lib, KVIKIO_STRINGIFY(cuFileGetVersion));
    int ver;
    CUfileError_t const error = GetVersion(&ver);
    if (error.err == CU_FILE_SUCCESS) { version = ver; }
  } catch (std::runtime_error const&) {
  }
#endif
```